File: codeforces/65.even_array/even_array.py
# Author: José Rodolfo (jric2002)
t = int(input())
while (t):
  n = int(input())
  min_mov = 0
  amount_zero = 0
  amount_one = 0
  #a = [int(e) for e in input().split(" ")]
  a = list(map(int, input().split(" ")))
  for i in range(0, n):
    if ((a[i] % 2) != (i % 2)):
      if ((a[i] % 2) == 0):
        amount_zero += 1
      else:
        amount_one += 1
  if (amount_zero != amount_one):
    min_mov = -1
  else:
    min_mov = amount_one
  print(min_mov)
  t -=1

File: codeforces/65.even_array/even_array.cpp
#include <iostream>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  unsigned short int t, n, a, amount_one, amount_zero;
  signed short int min_mov;
  cin >> t;
  while (t--) {
    min_mov = 0;
    amount_zero = 0;
    amount_one = 0;
    cin >> n;
    for (unsigned short int i = 0; i < n; i++) {
      cin >> a;
      if ((a % 2) != (i % 2)) {
        ((a % 2) == 0) ? amount_zero++ : amount_one++;
      }
    }
    (amount_zero != amount_one) ? min_mov = -1 : min_mov = amount_one;
    cout << min_mov << endl;
  }
  return 0;
}